Source/LibationSearchEngine/LuceneExtensions.cs

```diff
@@ -10,21 +10,58 @@ namespace LibationSearchEngine
 {
     // field names are case specific and, due to StandardAnalyzer, content is case INspecific
     internal static class LuceneExtensions
-    {
-        internal static void AddRaw(this Document document, string name, string value)
-            => document.Add(new Field(name, value, Field.Store.YES, Field.Index.NOT_ANALYZED));
+	{
+		internal static void AddAnalyzed(this Document document, string name, string value)
+		{
+			if (value is not null)
+				document.Add(new Field(name.ToLowerInvariant(), value, Field.Store.YES, Field.Index.ANALYZED));
+		}        
 
-        internal static void AddAnalyzed(this Document document, string name, string value)
-        {
-            if (value is not null)
-                document.Add(new Field(name.ToLowerInvariant(), value, Field.Store.YES, Field.Index.ANALYZED));
-        }
+        internal static void RemoveRule(this Document document, IIndexRule rule)
+		{
+			// fields are key value pairs. MULTIPLE FIELDS CAN POTENTIALLY HAVE THE SAME KEY.
```

Comment:
Thank you for retaining comments like these. They are hard won knowledge and I find they help considerably. Esp times like now years later when I've long since forgotten this gotcha